File: service/src/main/java/net/huray/backend/minuting/entity/MinutesEntity.java
package net.huray.backend.minuting.entity;

import static javax.persistence.GenerationType.IDENTITY;

import java.util.ArrayList;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import net.huray.backend.minuting.entity.common.BaseDateTimeEntity;

@Getter
@Entity(name = "minutes")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class MinutesEntity extends BaseDateTimeEntity {

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;

    private String title;

    private String contents;

    public MinutesEntity(String title, String contents) {
        this.title = title;
        this.contents = contents;
    }

    @OneToMany(mappedBy = "minutes")
    private List<MinutesAttendee> attendees = new ArrayList<>();

}

File: service/src/main/kotlin/net/huray/backend/minuting/entity/MinutesEntity.kt
package net.huray.backend.minuting.entity

import net.huray.backend.minuting.entity.common.BaseDateTimeEntity
import javax.persistence.*

@Entity(name = "minutes")
class MinutesEntity(
    var title: String = "",
    var contents: String = ""
) : BaseDateTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "minutes_id")
    var id: Long = 0L

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "space_id")
    lateinit var space: SpaceEntity

    @OneToMany(mappedBy = "minutes")
    var attendees: MutableList<MinutesAttendee> = mutableListOf()

    @OneToMany(mappedBy = "minutes")
    var comments: MutableList<CommentEntity> = mutableListOf()

    fun updateMinutes(title: String, contents: String) {
        this.title = title
        this.contents = contents
    }

    fun addComment(comment: CommentEntity) {
        this.comments.add(comment)
        comment.updateMinutes(this)
    }

}